geonode/metadata/handlers/group.py

```diff
@@ -68,8 +68,40 @@ def get_jsonschema_instance(self, resource, field_name, context, errors, lang=No
     def update_resource(self, resource, field_name, json_instance, context, errors, **kwargs):
         data = json_instance.get(field_name, None)
         id = data.get("id", None) if data else None
+        # retrieve the user which submitted the request
+        user = context.get("user", None)
+
+        if user is None:
+            self._set_error(
+                errors,
+                [field_name],
+                self.localize_message(
+                    context,
+                    "metadata_error_missing_owner",
+                    {"fieldname": field_name},
+                ),
+            )
+            return
+
         if id is not None:
-            gp = GroupProfile.objects.get(pk=id)
-            resource.group = gp.group
+
+            if user.is_superuser or user.is_staff:
+                allowed_groups = GroupProfile.objects.all()
+            else:
+                allowed_groups = GroupProfile.objects.filter(groupmember__user=user)
+
+            try:
+                gp = allowed_groups.get(pk=id)
+                resource.group = gp.group
+            except GroupProfile.DoesNotExist as e:
```

Comment:
In general is more efficient to use the [exists() rather then using the try/except block](https://stackoverflow.com/a/23567560/7597536). This can be converted in something like:

```python
gp = allowed_groups.filter(pk=id)
if gp.exists():
    resource.group = gp.group.first()
else:
    self._set_error(
        errors,
        [field_name],
        self.localize_message(
            context,
            "metadata_group_error_missing_group",
            {"fieldname": field_name, "exc": e},
        ),
    )
```